### ISSUE TITLE
const should be considered a bad part.

### ISSUE THREAD

**Issue by Skateside** [douglascrockford/JSLint#275]
I think `const` should be considered a bad part. I see a lot of developers using it under the assumption that it grants immutability, which it doesn't.

A good example of this bad practice is something like this:

```js
const foo = {
    one: 1,
    two: 2,
    three: 3
};
```

In that example, there's nothing stopping the code doing something like `foo.four = 4` even though the `const` would imply that it shouldn't work.

It's a similar issue when people use it to define a function.

```js
const doSomething = () => true;
```

Again, there's nothing stopping me writing something like `doSomething.foo = 1`.

I think `const` can be useful for literals - strings, numbers, booleans, `null` or `undefined` (although I can't imagine why anyone would want a constant for `null` or `undefined`) but if JSLint can't detect that, I feel like it should just reject `const` in favour of `let`.

**Comment by douglascrockford** [comment]
This confusion is due to a fundamental weakness in the way we teach programming: programmers should understand the difference between variables and values. 'const' works on variables. 'Object.freeze()' works on values. JavaScript's syntax is wrong in both cases. 'const' implies 'constant', which is not what 'const' does. That was clearly a mistake. 'Object.freeze()' is too verbose, and 'freeze' suggests the possibility of 'unfreeze' or 'thaw', which should never be allowed.

'const' is consistent with the best practices of functional programming and should be encouraged. 'let' should be avoided as much as practical.
